src/Services/DataService/DataService.test.ts

```diff
@@ -1058,70 +1063,69 @@ describe("DataService", () => {
           stubGetStateObservable.restore();
         }
 
-        if (stubFetchAll) {
-          stubFetchAll.restore();
-        }
-
-        if (stubGetItemsSelector) {
-          stubGetItemsSelector.restore();
-        }
       });
 
-      it("should get the correct items by query", () => {
-        stubGetItemsSelector = stub(fakeService.selectors, "getItems")
-          .returns([fakeModels[0], fakeModels[1]]);
+      it("should return a QueryManager with the correct items by query", () => {
+        const observable = fakeService.getByQuery(query);
 
-        const itemsObservable = fakeService.getByQuery(query);
-
-        itemsObservable.subscribe((items) => {
-          expect(items[0]).to.deep.equal(fakeModels[0]);
-          expect(items[1]).to.deep.equal(fakeModels[1]);
+        observable.take(1).subscribe((queryManager) => {
+          expect(queryManager.items[0]).to.deep.equal(fakeModels[0]);
+          expect(queryManager.items[1]).to.deep.equal(fakeModels[1]);
         });
       });
 
-      it("should get the correct items by their Ids and cache them for future requests by those Ids", () => {
+      it("should store the observable and return the same observable when given the same QueryBuilder", () => {
         const itemObservable = fakeService.getByQuery(query);
         const itemObservable2 = fakeService.getByQuery(query);
 
         expect(itemObservable).to.equal(itemObservable2);
       });
 
-      it("should not call BaseService.getStateObservable when using cached Observable by Ids", () => {
+      it("should not call BaseService.getStateObservable when using a cached Observable", () => {
         stubGetStateObservable = stub(BaseService, "getStateObservable").returns(state$);
-        const itemData = fakeModels;
 
         fakeService.getByQuery(query);
         fakeService.getByQuery(query);
 
-        expect(stubGetStateObservable).to.have.property("callCount").to.equal(1);
+        expect(stubGetStateObservable)
+          .to.have.property("callCount")
+          .to.equal(1);
       });
 
       it("should create a fetchAll action with the proper payload", () => {
-        stubFetchAll = stub(fakeService.actions, "fetchAll").returns({ invoke: spy() });
-
+        const observable = fakeService.getByQuery(query);
         fakeService.getByQuery(query);
 
-        expect(stubFetchAll.firstCall.args[0]).to.deep.equal(query);
+        observable.take(1).subscribe((queryManager) => {
+          expect(queryManager).to.deep.equal(query);
+        });
       });
 
       it("should not invoke the fetchAll action with the proper parameters if the requested Ids are already in the cache", () => {
-        const invokeSpy = spy();
-        stubFetchAll = stub(fakeService.actions, "fetchAll").returns({ invoke: invokeSpy });
+        const observable = fakeService.getByQuery(query);
 
-        fakeService.getByQuery(query);
         fakeService.getByQuery(query);
```

Comment:
I dont believe observables get fired off if they are not subscribed to, so I don't think this line does anything

Reply:
take(2) may do the trick when you have the test up and working expectedly